feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.ts

```diff
@@ -100,6 +101,33 @@ export class CheckoutDeliveryModeComponent {
     protected activeCartFacade: ActiveCartFacade
   ) {}
 
+  onKeydown(code: string | undefined, event: Event): void {
+    if (
+      !(
+        (<KeyboardEvent>event).key === 'Enter' ||
+        (<KeyboardEvent>event).key === ' '
+      )
+    )
+      return;
+
+    event.preventDefault();
+    const lastFocusedId = (<HTMLElement>event.target)?.id;
+    this.changeMode(code);
+
+    this.isUpdating$
+      .pipe(
+        filter((isUpdating) => !isUpdating),
+        take(1)
+      )
+      .subscribe(() => {
+        setTimeout(() => {
```

Comment:
Is there some way we can achieve the goal without `setTimeout`? It is known to cause issues with SSR and testing in cypress sometimes (due to delays between actions). Not critical since this is during a key event but it would be preferable.

https://sap.github.io/spartacus-docs/3.x/server-side-rendering-coding-guidelines/#working-with-timeouts

Reply:
I'm not happy with this either, unfortunately I could not get it to work without using `setTimeout`. I've tried a couple of things but could not get it to work.